### PR TITLE
Initial work for file-vault

### PR DESCRIPTION
**Pull request by easternbloc**

### CODE REVIEW

File: controllers/file.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+AWS.config.update({
+  accessKeyId: config.aws.accessKeyId,
+  secretAccessKey: config.aws.secretAccessKey,
+  region: config.aws.region,
+  signatureVersion: config.aws.signatureVersion
+});
+
+const s3 = new AWS.S3();
+
+router.post('/', upload.single('document'), clamAV, s3Upload, deleteUpload, (req, res, next) => {
```
Comment (joefitter) [inline controllers/file.js]: personally I prefer functions to be declared before they are used, thought our lint kicked off if you define them after?

File: controllers/file.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    });
+  }
+};
+
+function s3Upload (req, res, next) {
+  const params = {
+    Bucket: config.aws.bucket,
+    Key: req.file.filename,
+    Body: fs.createReadStream(req.file.path),
+    ServerSideEncryption: 'aws:kms',
```
Comment (joefitter) [inline controllers/file.js]: should this be taken from config?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  "license": "MIT",
+  "dependencies": {
+    "aws-sdk": "^2.7.21",
+    "churchill": "^0.2.0",
+    "express": "^4.14.0",
+    "hof-logger": "^1.0.0",
+    "moment": "^2.17.1",
+    "multer": "^1.2.1",
+    "request": "^2.79.0",
+    "winston": "^1.1.2"
```
Comment (joefitter) [inline package.json]: LINTLINTLINT

File: package.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  "devDependencies": {
+    "mocha": "^3.2.0",
+    "nock": "^9.0.2",
+    "should": "^11.1.2",
+    "supertest": "^2.0.1"
```
Comment (joefitter) [inline package.json]: LINTLINTLINT

**Comment by joefitter** [comment]
can we add .editorconfig too please? 🏩

File: controllers/file.js (hunk 1 above)
Comment (joefitter) [inline controllers/file.js]: ah, no lint

File: README.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# File-vault a RESTful service to store and retrieve files
+
+File-vault is a simple REST service that allows for POSTing of file to an S3 bucket. Upon a successful virus-scan the service will return with a URL that can be used to retrieve the file.
+
```
Comment (JoeChapman) [inline README.md]: How does one use it?

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      - AWS_ACCESS_KEY_ID=aws-access-key-id
+      - AWS_SECRET_ACCESS_KEY=aws-secret-access-key
+      - AWS_REGION=eu-west-1
+      - AWS_SIGNATURE_VERSION=v4
+      - AWS_BUCKET=aws-bucket
+    ports:
+      - "3000:3000"
+
+  nginx-proxy:
+    image: quay.io/ukhomeofficedigital/nginx-proxy:v1.5.1
```
Comment (JoeChapman) [inline docker-compose.yml]: latest is 2.1.0.0

**Comment by lennym** [comment]
I have a general preference for using `Error`s for errors, but I'm not your dad.

File: docker-compose.yml (hunk 6 above)
Comment (easternbloc) [inline docker-compose.yml]: it's in compose meh

**Pull request merged**